ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/JsonRpcHttpService.java

```diff
@@ -255,6 +260,50 @@ private void validateConfig(final JsonRpcConfiguration config) {
     return resultFuture;
   }
 
+  private HttpServerOptions getHttpServerOptions() {
+    return getHttpServerOptionsWithTls()
+        .setHost(config.getHost())
+        .setPort(config.getPort())
+        .setHandle100ContinueAutomatically(true);
+  }
+
+  private HttpServerOptions getHttpServerOptionsWithTls() {
+    final HttpServerOptions httpServerOptions = new HttpServerOptions();
+    if (!isTlsConfigurationEnabled()) {
+      return httpServerOptions;
+    }
+
+    final TlsConfiguration tlsConfiguration = config.getTlsConfiguration().get();
+    httpServerOptions
+        .setSsl(true)
+        .setPfxKeyCertOptions(storeToPfxOptions.apply(tlsConfiguration.getKeyStore()));
+
+    if (isClientAuthenticationRequired()) {
+      final Path knownClientsFile = tlsConfiguration.getKnownClientsFile().get();
+      httpServerOptions
+          .setClientAuth(ClientAuth.REQUIRED)
+          .setTrustOptions(VertxTrustOptions.whitelistClients(knownClientsFile));
+    }
+
+    return httpServerOptions;
+  }
+
+  private Function<TlsStoreConfiguration, PfxOptions> storeToPfxOptions =
+      tlsStoreConfiguration ->
+          new PfxOptions()
+              .setPath(tlsStoreConfiguration.getStorePath())
+              .setPassword(tlsStoreConfiguration.getStorePassword());
+
+  private boolean isTlsConfigurationEnabled() {
```

Comment:
nit: Not sure where this should go - but maybe it doesn't belong here - maybe it could be a better representation on the config object directly?
Wondering if the naming can be reduced - just "isTlsEnabled"?
Same thoughts go for function below

Reply:
It can be moved to `JsonRpcConfiguration` because thats where are optionally storing TlsConfiguration object.